controllers/teleport_master_controller/teleport_master_controller.cpp

```diff
@@ -0,0 +1,139 @@
+#include <fstream>
+#include <iostream>
+#include <random>
+#include <webots/Supervisor.hpp>
+#include <yaml-cpp/yaml.h>
+
+int main(int argc, char** argv) {
+
+    // Make sure we have the command line arguments we need. At a minimum we should have the def argument
+    // of the robot supervisor. Other arguments are def arguments of robots that should be controlled by
+    // this supervisor
+    if (argc < 2) {
+        std::cerr << "Usage: " << argv[0] << " <DEF>" << std::endl;
+        return EXIT_FAILURE;
+    }
+    // Load self-def argument which will be used to identify the robot using the webots getFromDef function
+    std::string def = argv[1];
+
+    // Load def arguments of other robots in the field into a vector
+    std::vector<webots::Node*> otherRobotsNodes;
+    // Only one supervisor instance can exist in a single controller. As such the same one will be used
+    // every time
+    webots::Supervisor supervisor = webots::Supervisor();
+    for (int i = 2; i < argc; i++) {
+        // Load the Node of the robot by def argument and add to vector
+        otherRobotsNodes.emplace_back(supervisor.getFromDef(argv[i]));
+    }
+
+    // AxisAngle rotations will be read from a config file and saved here
+    std::vector<std::array<double, 4>> rotations;
+    double minDistance;
+    double zHeight;
+    // Load config file
+    try {
+        YAML::Node config = YAML::LoadFile("config.yaml");
+        rotations         = config["rotations"].as<std::vector<std::array<double, 4>>>();
+        minDistance       = config["minDistance"].as<double>();
+        zHeight           = config["zHeight"].as<double>();
+    }
+    catch (const YAML::BadFile& e) {
+        std::cerr << e.msg << std::endl;
+        return 1;
+    }
+    catch (const YAML::ParserException& e) {
+        std::cerr << e.msg << std::endl;
+        return 2;
+    }
+
+    webots::Node* tempNode = supervisor.getFromDef("test");
+    const double xSize     = tempNode->getField("xSize")->getSFFloat();
+    const double ySize     = tempNode->getField("ySize")->getSFFloat();
+
+    // Get the time step of the current world.
+    int timeStep = int(supervisor.getBasicTimeStep());
+
+    // Generate random seed
+    std::random_device rd;   // Will be used to obtain a seed for the random number engine
+    std::mt19937 gen(rd());  // Standard mersenne_twister_engine seeded with rd()
+
+    // Generate distributions for random number systems in the following loop
+    std::uniform_real_distribution<> xDistrib(-xSize * 0.5, xSize * 0.5);
+    std::uniform_real_distribution<> yDistrib(-ySize * 0.5, ySize * 0.5);
+    std::uniform_real_distribution<> rotDistrib(0.0, rotations.size() - 1.0);
+
+    while (supervisor.step(timeStep) != -1) {
+
+        // Declare a vector of positions that will be saved as they are randomly generated, to be later
+        // applied to each robot
+        std::vector<std::array<double, 3>> positions = std::vector<std::array<double, 3>>();
```

Comment:
No need to do this. You are effectively calling the default constructor (twice, I think) and then copying that into `positions`. Without the extra bit you are just calling the default constructor.
```suggestion
        std::vector<std::array<double, 3>> positions;
```

Reply:
Oh tright I didn't realise that was what that did. I was treating it like a primitive array where you gotta initialise it first

Reply:
If it was a primitive array that you had to initialise first this is still the wrong way of doing it :P
Also don't use primitive arrays in c++ nowadays, use `std::array<type, count>` and initialise it by doing
```cpp
std::array<int, 5> thing{};
```
Which will set it to all 0s (or all default constructed elements if they are not primitives)

Reply:
Oh my mind was in java town when I said that. But yea Kip got me using the std::array last time, it's much nicer :D